### PR TITLE
add jwt as an authorization mode

### PR DESCRIPTION
**Pull request by duytiennguyen-okta**
## Summary

Fixes #

## Type of PR

- [ ] Bug Fix (non-breaking fixes to existing functionality)
- [ ] New Feature (non-breaking changes that add new functionality)
- [ ] Documentation update
- [ ] Test Updates
- [ ] Other (Please describe the type)

## Test Information

- [ ] My PR required test updates 

Go Version:
Os Version:
OpenAPI Spec Version:

## Signoff
- [ ] I have submitted a CLA for this PR
- [ ] Each commit message explains what the commit does
- [ ] I have updated documentation to explain what my PR does
- [ ] My code is covered by tests if required
- [ ] I ran `make fmt` on my code
- [ ] I did not edit any automatically generated files

### CODE REVIEW

File: okta/requestExecutor.go [hunk 1]
```diff
@@ -240,7 +418,7 @@ func (re *RequestExecutor) NewRequest(method string, url string, body interface{
 	re.binary = false
 	re.headerAccept = "application/json"
 	re.headerContentType = "application/json"
-
+	fmt.Println("433333", req.Header)
```
Comment (bretterer) [inline okta/requestExecutor.go:421]: Should remove this

File: README.md [hunk 2]
```diff
@@ -1055,6 +1055,8 @@ ctx, client, err := okta.NewClient(ctx,
 
 ```
 
+### OAuth 2.0 With JWT Key
+
```
Comment (monde) [inline README.md:1059]: Is there missing documentation for this section?

File: okta/config.go [hunk 3]
```diff
@@ -134,6 +135,12 @@ func WithToken(token string) ConfigSetter {
 	}
 }
 
+func WithClientAssertion(clientAssertion string) ConfigSetter {
```
Comment (monde) [inline okta/config.go:138]: Should add an entry in the README where all the other `With*` config helpers are explained.

Comment (monde) [reply okta/config.go:138]: https://github.com/okta/okta-sdk-golang#configuration-setter-object

File: okta/okta.go [hunk 4]
```diff
@@ -150,6 +150,19 @@ func (c *Client) GetConfig() *config {
 	return c.config
 }
 
+func (c *Client) SetConfig(conf ...ConfigSetter) (err error) {
```
Comment (monde) [inline okta/okta.go:153]: I'm not sure sure the best way to document this is. Just on the public method? Also in the readme?

**Review by monde — COMMENTED** [review]
Aren't there hbs files that generated these changes?

**Review by monde — APPROVED** [review]

**Pull request merged**